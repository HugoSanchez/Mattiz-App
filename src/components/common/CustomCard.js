import React from 'react';
import { View } from 'react-native';

// Props:
// @elevated: if additional prop styles if true, else null.

const CustomCard = props => {
    return (
        <View style={[ 
                styles.container, 
                props.style, 
                props.elevated ? {
                    shadowColor: '#000',
                    shadowOffset: { width: 0, height: 3 },
                    shadowOpacity: 0.29,
                    shadowRadius: 4.65,
                    elevation: 7 } 
                    : 
                    null 
                ]}
            >
            {
                props.children
            }
        </View>
    );
}

const styles = {
    container: {
        alignSelf: 'stretch',
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 15,
        backgroundColor: '#FDFFFE'
    }
};

export default CustomCard;