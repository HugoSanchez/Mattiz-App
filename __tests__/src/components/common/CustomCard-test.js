import 'react-native'
import React from 'react'
import CustomCard from '../../../../src/components/common/CustomCard'
import {render, fireEvent} from '@testing-library/react-native'

test('Renders with title', () => {
	const {container} = render(<CustomCard />)
	expect(container.children[0]).toMatchSnapshot()
})
