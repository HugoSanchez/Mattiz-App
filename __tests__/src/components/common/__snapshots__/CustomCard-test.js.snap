// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#FDFFFE",
        "borderRadius": 15,
        "justifyContent": "center",
        "margin": 5,
      },
      undefined,
      null,
    ]
  }
/>
`;
